### PR TITLE
Make "placeholder" of ODBC configurable in UI

### PR DESCRIPTION
**Pull request by jarnedemunter1**
Parametrize the "placeholder" variable in the ODBC hook, making it configurable from the connection defined in the UI.
For some databases, the placeholder is not the standard "%s" as defined earlier.
By making it configurable in the connection properties, users can easily pick the placeholder of their specific database.
This makes the generic_transfer operator more flexible.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/common/sql/hooks/sql.py [hunk 1]
```diff
@@ -144,7 +144,7 @@ class DbApiHook(BaseHook):
     # Override with db-specific query to check connection
     _test_connection_sql = "select 1"
     # Override with the db-specific value used for placeholders
-    placeholder: str = "%s"
+    _placeholder: str = "%s"
```
Comment (hussein-awala) [inline airflow/providers/common/sql/hooks/sql.py]: This could be moved to `__init__` since it's no longer a class attribute.

**Review by hussein-awala — APPROVED** [review]
A small nit, otherwise LGTM

**Review by potiuk — CHANGES_REQUESTED** [review]
We should only  allow fixed set of placeholders - not a "Free placeholder" = only alow "%s" or "?" - nothing else. 

The problem with such an input provided by the user who can configure a connection from the UI, that by allowing "ANY" values, it opens up the doors to all kind of vulnerabilities - similar to SQL  INJECTION. 

So we should always sanitize any input passed down from the user when such conection is defined via UI. 

While we cannot prevent existing cases and some kinds of "inherent" vulnerabilities in some connection types, adding a gateway to such "user input" (even if we do not know if it can be exploited) is dangerous.

In this case I think "?" and "%s" are the only "reasonable" values placeholder can take and we should validate if user value is exactly one of those..

 Maybe there are others that we might find and add, but adding a possibility of using ANY placeholder provided by the user here is too dangerous IMHO.

**Comment by hussein-awala** [comment]
> We should only allow fixed set of placeholders - not a "Free placeholder" = only alow "%s" or "?" - nothing else.
> 
> The problem with such an input provided by the user who can configure a connection from the UI, that by allowing "ANY" values, it opens up the doors to all kind of vulnerabilities - similar to SQL INJECTION.
> 
> So we should always sanitize any input passed down from the user when such conection is defined via UI.
> 
> While we cannot prevent existing cases and some kinds of "inherent" vulnerabilities in some connection types, adding a gateway to such "user input" (even if we do not know if it can be exploited) is dangerous.
> 
> In this case I think "?" and "%s" are the only "reasonable" values placeholder can take and we should validate if user value is exactly one of those..
> 
> Maybe there are others that we might find and add, but adding a possibility of using ANY placeholder provided by the user here is too dangerous IMHO.

Makes sense! We can achieve this easily by adding a list of whitelisted placeholders in each provider, and checking if the provided placeholder exists in this list.

**Comment by jarnedemunter1** [comment]
We implemented the change by adding the fixed set of default values to the ODBC provider. The other providers already used a "standard" placeholder. This first example can be used for other providers should it be necessary in the future

**Review by potiuk — APPROVED** [review]

**Pull request merged**